frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/services/MfaResetIPVAuthorizationServiceTest.java

```diff
@@ -0,0 +1,239 @@
+    private final JwtService jwtService = mock(JwtService.class);
+    private final NowHelper.NowClock nowClock = TestClockHelper.getInstance();
+    private final TokenService tokenService = mock(TokenService.class);
+    private final ConfigurationService configurationService = mock(ConfigurationService.class);
+    private final RedisConnectionService redisConnectionService =
+            mock(RedisConnectionService.class);
+    private final AuditService auditService = mock(AuditService.class);
+    private final AuditContext auditContext = mock(AuditContext.class);
+    private final CloudwatchMetricsService cloudwatchMetricsService =
+            mock(CloudwatchMetricsService.class);
+    private final JWTClaimsSet TEST_JWT_CLAIMS = constructTestClaimSet();
+    private final MfaResetIPVAuthorizationService mfaResetIPVAuthorizationService =
+            new MfaResetIPVAuthorizationService(
+                    configurationService,
+                    nowClock,
+                    jwtService,
+                    tokenService,
+                    redisConnectionService,
+                    auditService,
+                    cloudwatchMetricsService);
+    private SignedJWT TEST_SIGNED_JWT;
+    private EncryptedJWT TEST_ENCRYPTED_JWT;
+
+    @BeforeEach
+    void testSetup() throws URISyntaxException, ParseException {
+        when(tokenService.generateStorageTokenForMfaReset(any()))
+                .thenReturn(new BearerAccessToken(TEST_STORAGE_TOKEN));
+        when(configurationService.getMfaResetJarSigningKeyAlias()).thenReturn(TEST_KEY_ALIAS);
+        when(configurationService.getStorageTokenClaimName()).thenReturn(TEST_STORAGE_TOKEN_CLAIM);
+        when(configurationService.getMfaResetCallbackURI())
+                .thenReturn(new URI(TEST_MFA_CALLBACK_URI));
+        when(configurationService.getIPVAuthEncryptionPublicKey()).thenReturn(TEST_PUBLIC_KEY);
+        when(configurationService.getIPVAuthorisationClientId())
+                .thenReturn(TEST_IPV_AUTH_CLIENT_ID);
+        when(configurationService.getAuthIssuerClaim()).thenReturn(TEST_ISSUER_CLAIM);
+        when(configurationService.getIPVAuthorisationURI())
+                .thenReturn(new URI(TEST_IPV_AUTHORIZE_URI));
+        when(configurationService.getSessionExpiry()).thenReturn(TEST_SESSION_EXPIRY);
+        when(configurationService.getIPVAudience()).thenReturn(TEST_AUDIENCE_CLAIM);
+        TEST_SIGNED_JWT =
+                new SignedJWT(
+                        TEST_ENCODED_JWS_HEADER,
+                        Base64URL.encode(TEST_JWT_CLAIMS.toString()),
+                        TEST_ENCODED_JWS_SIGNATURE);
+        TEST_ENCRYPTED_JWT =
+                new EncryptedJWT(
+                        TEST_JWE_FIRST_PART,
+                        TEST_JWE_SECOND_PART,
+                        TEST_JWE_THIRD_PART,
+                        TEST_JWE_FOURTH_PART,
+                        TEST_JWE_FIFTH_PART);
```

Comment:
If we don't care about the value could we just use `mock(EncryptedJWT.class)`?

Reply:
So I was mocking the return value so we could assert that the authorise request URL included the mocked return value, not sure if this is worth it though

Reply:
If we don't care about the structure of the value we could use a mock and configure it to return a specific string when we call `toString` or whatever on it?